### DISCUSSION TITLE
Playfield borders scaling to circle size

### DISCUSSION THREAD

**Discussion by vernonlim** [ppy/osu#13167] [Ideas] [answered]
As mentioned in #9484, the playfield border currently is significantly smaller than what players use, due to being at the limits of editor object placement, not the limits of where a circle can be.

To fix this, playfield borders could scale to circle size. What I mean is that the limits of where a circle can be is calculated based on the CS, and then the border is scaled to those limits. This is actually already implemented in the third party client McOsu.

(in McOsu)
border in a CS10 map:
![screenshot96](https://user-images.githubusercontent.com/25032854/96745291-9191ca00-13f8-11eb-841b-2032233fec6a.png)

border with CS4 on the same map:
![screenshot98](https://user-images.githubusercontent.com/25032854/96745383-a8d0b780-13f8-11eb-9878-325da57929fd.png)

In my opinion this is the best form of playfield borders (better than the static ones in skins), although I am not sure how hard it is to implement.

**Comment by bdach** [comment]
Applying the adjustment as proposed would probably end up being a simple padding/margin set. The bigger issue here is how to read/calculate the adjustment sanely - we already have quite a few distinct places where the CS formula is used locally. Should probably not be the case with this if at all possible.

**Reply by bdach** [reply, marked as answer]
https://github.com/ppy/osu/pull/28504
